mlflow/pytorch/__init__.py

```diff
@@ -325,6 +376,34 @@ def save_model(
         f.write(pickle_module.__name__)
     # Save pytorch model
     model_path = os.path.join(model_data_path, _SERIALIZED_TORCH_MODEL_FILE_NAME)
+
+    if requirements_file:
+        if not isinstance(requirements_file, str):
+            raise TypeError("Path to requirements file should be a string")
+
+        with TempDir() as tmp_requirements_dir:
```

Comment:
The logic here seems correct, but just wondering - do people tend to log requirements.txt files from remote artifact stores, e.g. s3, as part of their pytorch models?

We could simplify this by requiring requirements.txt to be a file on the local filesystem. With the current logic, in the seemingly-common case where requirements.txt is in the local filesystem, we'll copy it to another temporary directory on the same local filesystem, which seems a bit redundant

Reply:
From offline discussion with @chauhang:

It varies based on the files. The requirements.txt will generally be local or from github. The vocab files will be from a shared storage like S3. Other files will be very model specific so we won't know where they will come from. Hence we need support for remote download of the files if they are not local

Thus let's stick with the logic as is for now - we can iterate on it later if we need to optimize for whatever reason